create.ps1

```diff
@@ -165,118 +127,148 @@ if ($false -eq $actionContext.Configuration.setDepartment) {
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        if ([string]::IsNullOrEmpty($correlationField)) {
+            throw "Correlation is enabled but not configured correctly."
+        }
+    
+        if ([string]::IsNullOrEmpty($correlationValue)) {
+            throw "The correlation value for [$correlationField] is empty. This is likely a mapping issue."
+        }
+    }
+    else {
+        Write-Warning "Correlation is disabled."
+    }
+    #endregion Verify correlation configuration and properties
 
-        $outputContext.AuditLogs.Add([PSCustomObject]@{
-                # Action  = "" # Optional
-                Message = "Error creating authorization headers. Error Message: $($errorMessage.AuditErrorMessage)"
-                IsError = $true
-            })
+    #region Create access token
+    $actionMessage = "creating access token"
 
-        # Throw terminal error
-        throw
+    $createAccessTokenBody = @{
+        grant_type                = "client_credentials"
+        client_id                 = $actionContext.Configuration.clientId
+        client_secret             = $actionContext.Configuration.clientSecret
+        token_expiration_disabled = $false
     }
 
-    # Check if we should try to correlate the account
-    if ($actionContext.CorrelationConfiguration.Enabled) {
-        $correlationField = $actionContext.CorrelationConfiguration.accountField
-        $correlationValue = $actionContext.CorrelationConfiguration.accountFieldValue
+    $createAccessTokenSplatParams = @{
+        Uri             = "$($actionContext.Configuration.serviceAddress)/oauth/token"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/json"
+        UseBasicParsing = $true
+        Body            = ($createAccessTokenBody | ConvertTo-Json)
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
 
-        if ($correlationField -eq $null) {
-            Write-Warning "Correlation is enabled but not configured correctly."
-        }
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
 
-        # Check if current account exists and and verify if a user must be either [created] or [correlated]
-        try {
-            $filter = "$($correlationField) eq `"$($correlationValue)`""
-            Write-Verbose "Querying account that matches filter [$($filter)]"
-            $splatWebRequest = @{
-                Uri             = "$baseUrl/scim/users?filter=$([System.Uri]::EscapeDataString($filter))"
-                Headers         = $headers
-                Method          = 'GET'
-                ContentType     = "application/json;charset=utf-8"
-                UseBasicParsing = $true
-            }
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
```

Comment:
Dit zou ik vervolgens ook weghalen denk ik? (geld ook voor elk script)

Reply:
Correct, is weggehaald.